src/gui/mainwindow.cpp

```diff
@@ -317,11 +315,11 @@ MainWindow::MainWindow(QWidget *parent)
     connect(m_ui->actionUseAlternativeSpeedLimits, &QAction::triggered, this, &MainWindow::toggleAlternativeSpeeds);
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
-    m_programUpdateTimer = new QTimer(this);
-    m_programUpdateTimer->setInterval(60 * 60 * 1000);
-    m_programUpdateTimer->setSingleShot(true);
-    connect(m_programUpdateTimer, &QTimer::timeout, this, &MainWindow::checkProgramUpdate);
-    connect(m_ui->actionCheckForUpdates, &QAction::triggered, this, &MainWindow::checkProgramUpdate);
+    connect(m_ui->actionCheckForUpdates, &QAction::triggered, this, [this]() { checkProgramUpdate(true); });
```

Comment:
What happens if user trigger it while ProgramUpdater checking for updates?

Reply:
It is not possible, the menu item is disabled after selection (and re-enabled later).